manim/mobject/mobject.py

```diff
@@ -1764,7 +1782,7 @@ def set_colors_by_radial_gradient(
         )
         return self
 
-    def set_submobject_colors_by_gradient(self, *colors):
+    def set_submobject_colors_by_gradient(self, *colors: Iterable[ParsableManimColor]):
```

Comment:
Star-args are expected by mypy (and the relevant PEP) to be typed by the type of a single item, so we should remove the `Iterable` here

Reply:
See my review here: https://github.com/ManimCommunity/manim/pull/3086#discussion_r1081870227. As I now have rights to push on the branch I'll probably fix it this weekend along with some fixes on the work done from alembcke